dlpy/layers.py

```diff
@@ -555,39 +563,28 @@ def __init__(self, n_filters, width = None, height = None, stride = 1, name = No
     def output_size(self):
         if self._output_size is None:
             # calculate output according to specified padding
-            if self.config['padding'] is not None:
-                out_w = (self.src_layers[0].output_size[0] -
-                         self.config['width'] + 2*self.config['padding']) // self.config['stride'] + 1
-                out_h = (self.src_layers[0].output_size[1] -
-                         self.config['height'] + 2*self.config['padding']) // self.config['stride'] + 1
+            if self.padding != (None, None):
+                out_h = (self.src_layers[0].output_size[0]-self.config['height'] + 2*self.padding[0])//self.stride[0]+1
+                out_w = (self.src_layers[0].output_size[1]-self.config['width'] + 2*self.padding[1])//self.stride[1]+1
             else:
                 import math
                 # same padding
-                out_w = math.ceil(self.src_layers[0].output_size[0] / self.config['stride'])
-                out_h = math.ceil(self.src_layers[0].output_size[1] / self.config['stride'])
-
-                # if either padding_height or padding_width are specified
-                if self.config['padding_width'] is not None:
-                    out_w = (self.src_layers[0].output_size[0] -
-                             self.config['width'] +
-                             2*self.config['padding_width']) // self.config['stride'] + 1
-                if self.config['padding_height'] is not None:
-                    out_h = (self.src_layers[0].output_size[1] -
-                             self.config['height'] +
-                             2*self.config['padding_height']) // self.config['stride'] + 1
-            self._output_size = (int(out_w), int(out_h), int(self.config['n_filters']))
+                out_h = math.ceil(self.src_layers[0].output_size[0]/self.stride[0])
```

Comment:
self.stride[0] cannot be 0 right?

Reply:
It is not easy to be zero, as I have a place to prevent it to be 0.        
`        if parameters['width'] == 0 and parameters['height'] == 0:
            self.stride = _pair(1)`
But there is a chance to make it zero if you do Pooling(width=0, height=0, stride=0). But I don't think people will do so. Anyway, I will handle the case.